zerver/lib/mention.py

```diff
@@ -105,9 +108,15 @@ def get_full_name_info_list(self, user_filters: List[UserFilter]) -> List[FullNa
                 )
             )
 
+            possible_mention_user_ids = [row.id for row in rows]
+            inaccessible_user_ids = get_inaccessible_user_ids(
+                possible_mention_user_ids, message_sender
+            )
```

Comment:
I this doesn't work correctly if the user they are trying to mention is in the `user_filters` cache.

Reply:
Ohh, forgot to comment on this.

For sending messages through API and most of the other cases, we always create a new `MentionBackend` object for each message so `user_cache` is initialized for each message. This happens for most of the cases except when we send bot notifications for stream and group subscriptions.

Reply:
I see, is another framing that the cache is never reused with multiple senders?